### PR TITLE
fix: matrix limit not respected

### PR DESCRIPTION
**Pull request by TheGreatRefrigerator**
### Pull Request Checklist

- [x] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [x] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

### Information about the changes
- Key functionality added: fixed matrix 'maximum_route' limit for explicit 'all' in destiantions and/or sources
- Reason for change: matrix requests could circumvent the route limit as 'all' was counted as single entry during the check

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/run-instance/configuration/
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

### CODE REVIEW

File: ors-api/src/main/java/org/heigit/ors/api/services/MatrixService.java [hunk 1]
```diff
@@ -55,8 +56,10 @@ public org.heigit.ors.matrix.MatrixRequest convertMatrixRequest(MatrixRequest ma
                 endpointsProperties.getMatrix().getMaximumVisitedNodes(),
                 endpointsProperties.getMatrix().getUTurnCost());
 
-        int numberOfSources = matrixRequest.getSources() == null ? matrixRequest.getLocations().size() : matrixRequest.getSources().length;
-        int numberODestinations = matrixRequest.getDestinations() == null ? matrixRequest.getLocations().size() : matrixRequest.getDestinations().length;
+        String[] sources = matrixRequest.getSources();
```
Comment (TheGreatRefrigerator) [inline ors-api/src/main/java/org/heigit/ors/api/services/MatrixService.java:59]: is it better to inline this?

**Comment by aoles** [comment]
Is it actually desired to allow passing a String value of `"all"` as a valid value to the `sources`/`destination` parameter? To be honest, I didn't even know this was possible. I always understood these parameters as integer arrays, which if not provided defaulted to all locations, i.e. arrays of [0, n-1] where n is the number of provided `locations`. I'm not sure whether mixing different types such as strings and integers within one parameter is a good idea. So maybe the actual issue here is about allowing passing "all" as a valid parameter value, which to me sounds a bit obsolete considering that the default is to take all locations anyway.

**Comment by TheGreatRefrigerator** [comment]
> Is it actually desired to allow passing a String value of `"all"` as a valid value to the `sources`/`destination` parameter? To be honest, I didn't even know this was possible. I always understood these parameters as integer arrays, which if not provided defaulted to all locations, i.e. arrays of [0, n-1] where n is the number of provided `locations`. I'm not sure whether mixing different types such as strings and integers within one parameter is a good idea. So maybe the actual issue here is about allowing passing "all" as a valid parameter value, which to me sounds a bit obsolete considering that the default is to take all locations anyway.

Can we discuss this in a separate issue please. Currently you can circumvent the config limit. That's fixed with the PR. Being able to set `all` has been this way at least since 2018. The whole array is read in as strings and converted afterwards...
Of course that's not conceptually ideal, but i don't think that's relevant for this PR.

File: ors-api/src/main/java/org/heigit/ors/api/services/MatrixService.java (hunk 1 above)
Comment (aoles) [reply ors-api/src/main/java/org/heigit/ors/api/services/MatrixService.java:59]: I like the current version better 👍

**Review by aoles — APPROVED** [review]
LGTM, cheers! 🌈

**Pull request merged**